pyccl/halos/profiles/profile_base.py

```diff
@@ -66,12 +76,11 @@ def get_normalization(self, cosmo, a, *, hmc=None):
             float: normalization factor of this profile.
         """
         def integ(M):
-            return self.fourier(cosmo=cosmo,
-                                k=hmc.precision['k_min'],
-                                M=M, a=a, mass_def=hmc.mass_def)
+            return self.fourier(cosmo, hmc.precision["k_min"], M, a)
         return hmc.integrate_over_massfunc(integ, cosmo, a)
         # TODO: CCLv3 replace by the below in v3 (profiles will all have a
         # default normalization of 1. Normalization will always be applied).
+        # NK: (cosmo, a) have to take None defaults in v3.
```

Comment:
what?

Reply:
We should be able to call `prof.get_normalization()` with no arguments in the degenerate case where the norm is 1. How it is now, it requires that `cosmo` and `a` are necessarily passed. It makes no sense to have `hmc` as optional and the other two as required arguments. So the signature should be `(self, cosmo=None, a=None, *, hmc=None)`.